app_dart/lib/src/foundation/utils.dart

```diff
@@ -89,8 +89,9 @@ Future<RepositorySlug> repoNameForBuilder(List<LuciBuilder> builders, String bui
 ///
 /// Only `enabled` luci builders will be returned.
 Future<List<LuciBuilder>> getRepoBuilders(HttpClientProvider branchHttpClientProvider, Logging log,
```

Comment:
The logic "fetch builders -> list files -> get filtered builders" is duplicated repetitively 3 times in luci_build_service and github_status_service. Since we always use filtered builders, it's better to move the logic into this function.

Bonus point is this can simply read `X_builders.json` from `commitSha` (including when it's `master`), regardless whether the PR contains a change to  `X_builders.json`.

Reply:
Moved all logics here as suggested (renamed function to `getLuciTryBuilders`), and enabled default reading from `commitSha` for all Try builders.